### PR TITLE
refactor(http): deprecated symbol used

### PR DESCRIPTION
**Pull request by SrdjanMilic**
Support for passing an error value to "throwError" method will be removed in v8 of rxjs. More info at https://rxjs.dev/api/index/function/throwError#throwerror

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Deprecated symbol used.

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by atscott — APPROVED** [review]

**Comment by dylhunn** [comment]
Can you rebase and push again to this branch? The CI doesn't seem to have run for some reason.

**Comment by SrdjanMilic** [comment]
Could please someone tell me where I could find info about "Style Guide 07-01" as it seems that failed test is all about that?

**Comment by atscott** [comment]
@SrdjanMilic This is the test that's failing:
https://github.com/angular/angular/blob/486cade596a57b7c146ae3aa269d69d7d828a6ab/aio/content/examples/http/e2e/src/app.e2e-spec.ts#L106-L110
It's in the example app that you modified in this PR. I tried to update the app based on a guess from what changed and pushed a fixup commit to your branch. We'll see if the test passes now.

File: aio/content/examples/http/src/app/config/config.service.ts [hunk 1]
```diff
@@ -82,8 +82,9 @@ export class ConfigService {
         `Backend returned code ${error.status}, body was: `, error.error);
     }
     // Return an observable with a user-facing error message.
-    return throwError(
-      'Something bad happened; please try again later.');
+    return throwError(() => {
+      new Error('Something bad happened; please try again later.');
+    });
```
Comment (markostanimirovic) [inline aio/content/examples/http/src/app/config/config.service.ts:87]: The error should be returned:

```suggestion
    return throwError(() => {
      return new Error('Something bad happened; please try again later.');
    });
```

This should fix the test 🙂

Comment (atscott) [reply aio/content/examples/http/src/app/config/config.service.ts:87]: :) That'll do it!

**Review by atscott — APPROVED** [review]

**Comment by SrdjanMilic** [comment]
Great! Thank you all for help provided.

**Comment by alxhub** [comment]
This PR was merged into the repository by commit bbababe5900ea8f4c8fccd88238f6fe08a2ceb63.

**Pull request closed without merging**